lib/sqlsync/src/reducer.rs

```diff
@@ -29,16 +29,29 @@ pub enum ReducerError {
 
     #[error(transparent)]
     Sqlite(#[from] rusqlite::Error),
+
+    #[error(transparent)]
+    External(Box<dyn std::error::Error + Send + Sync>),
 }
 
-type Result<T> = std::result::Result<T, ReducerError>;
+pub type Result<T> = std::result::Result<T, ReducerError>;
 type SqlResult<T> = std::result::Result<T, ErrorResponse>;
 
-pub struct Reducer {
+pub trait Reducer {
+    fn apply(&mut self, tx: &mut Transaction, mutation: &[u8]) -> Result<()>;
+}
+
+impl Reducer for WasmReducer {
+    fn apply(&mut self, tx: &mut Transaction, mutation: &[u8]) -> Result<()> {
+        WasmReducer::apply(self, tx, mutation)
+    }
+}
```

Comment:
Pretty stoked how clean this ended up being. If you have time, I would prefer to just move the WasmReducer apply impl into this impl, which probably means you'll need to make the reducer generic in local.rs (which looks like it's just a couple lines of changes a'la coordinator). No worries if you leave this, I can do it later.